src/devices/bus/hp9845_io/hp9845_io.cpp

```diff
@@ -74,7 +74,7 @@ void hp9845_io_card_device::flg_w(int state)
 void hp9845_io_card_device::install_readwrite_handler(read16_delegate rhandler, write16_delegate whandler)
 {
 	if (m_sys == nullptr) {
-		m_sys = dynamic_cast<hp9845b_state*>(&machine().root_device());
+		m_sys = dynamic_cast<hp9845_base_state*>(&machine().root_device());
```

Comment:
Making the assumption that the root device is the machine you want is pretty dangerous, especially as we're trying to move to the "everything is a device" model.  Are you sure you can't supply what you need through the slot interface somehow?